sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/joins.scala

```diff
@@ -43,10 +43,15 @@ object ReorderJoin extends Rule[LogicalPlan] with PredicateHelper {
    *
    * @param input a list of LogicalPlans to inner join and the type of inner join.
    * @param conditions a list of condition for join.
+   * @param leftPlans a list of LogicalPlans contained in the left join child.
+   * @param hintMap a map of relations to their corresponding hints.
```

Comment:
`a map of relations ...`, the map key is not `Seq[LogicalPlan]` anymore, shall we update this?

Reply:
Updated